### DISCUSSION TITLE
23.04.2021

### DISCUSSION THREAD

**Discussion by farag2** [farag2/Sophia-Script-for-Windows#162] [General]
# Before running the script read the [Before running](https://github.com/farag2/Windows-10-Sophia-Script#fire-before-running-fire) & [How to use](https://github.com/farag2/Windows-10-Sophia-Script#how-to-use)

## Sophia Script in Action

[![YT](https://img.youtube.com/vi/f529ucAipI8/0.jpg)](https://youtu.be/f529ucAipI8)

[![YT](https://img.youtube.com/vi/MiQ85tVXQQA/0.jpg)](https://youtu.be/MiQ85tVXQQA)

***

[![ko-fi](https://www.ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/Q5Q51QUJC)

<a href="https://yoomoney.ru/to/4100116615568835"><img src="https://yoomoney.ru/i/shop/iomoney_logo_color_example.png" width=220px height=46px></a>

***

<a href="https://t.me/SophiaNews"><img src="https://img.shields.io/badge/Sophia%20News-Telegram-blue?style=flat&logo=Telegram"></a>
<a href="https://t.me/Sophia_Chat"><img src="https://img.shields.io/badge/Sophia%20Chat-Telegram-blue?style=flat&logo=Telegram"></a>

<img src="https://img.shields.io/github/workflow/status/farag2/Windows-10-Sophia-Script/Build?label=GitHub%20Actions&logo=GitHub">

## Windows 10 2004 (20H1)/20H2 (2009) | LTSC

Diff from v5.10.1
[5.10.1...5.10.2](https://github.com/farag2/Windows-10-Sophia-Script/compare/5.10.1...5.10.2)

* Updated the descriptions;
* PowerShell 7 version;
  * CsWinRT updated up to 1.2.5.
* Minor changes. :feelsgood:
* The development focus shifted to [SophiApp](https://github.com/Sophia-Community/SophiApp) :rocket:

Always check hashes: `(Get-FileHash -Path "path/to/archive.zip" -Algorithm SHA256).Hash`

|              File                      |      SHA-256   |
|:--------------------------------------:|:-----------------------------:|
| Sophia.Script.LTSC.v5.2.2.zip          | 9D5B32BD5497E46E7AD054E1DA973E9A95BDECE4DA792CBD72CAA30632B69853|
| Sophia.Script.v5.10.1.PowerShell.7.zip |118090D539906CE1CAE982B658F7C3199ED582A4BDEDD6E9EDC951E5B37AAA6D|
| Sophia.Script.v5.10.2.zip              |82567817994E4573FD39801B804C4B9AA8FD9E1DC14450041E6BEE14545549CF|
| Sophia.Script.Wrapper.zip              |F9FE14E6388A1E96B8C067F184D93285C6C94E2474DAC4AC5E818F2C7C3732C1|

Learn more about David's [wrapper](https://github.com/farag2/Windows-10-Sophia-Script/tree/master/Wrapper) from [benchtweakgaming.com](https://benchtweakgaming.com/2020/11/12/windows-10-debloat-tool/).

[![YT](https://img.youtube.com/vi/f529ucAipI8/1.jpg)](https://www.youtube.com/watch?v=f529ucAipI8&t=946s)

<hr /><em>This discussion was created from the release <a href='https://github.com/farag2/Windows-10-Sophia-Script/releases/tag/5.10.2'>23.04.2021</a>.</em>

**Comment by ghost** [comment]
Now that you provide hashes, would you mind uploading an additional file named `SHA256SUM` or whatever in following releases?

PS: I notice you [haven't uploaded GPG keys to GitHub](https://github.com/farag2.gpg) so don't bother if you don't want PGP verification.

**Reply by farag2** [reply]
What this file should consist of? I'm not familiar with this.

**Reply by ghost** [reply]
My bad, I prefer the format like the following so I can simply check hashes by `shasum -c SHA256SUM` (PowerShell does not come with tools to *check* sum so I use [psutils](https://github.com/lukesampson/psutils) as an approximation of [shasum](https://linux.die.net/man/1/shasum)).

    Sophia.Script.LTSC.v5.2.2.zip 9D5B32BD5497E46E7AD054E1DA973E9A95BDECE4DA792CBD72CAA30632B69853
    Sophia.Script.v5.10.1.PowerShell.7.zip 118090D539906CE1CAE982B658F7C3199ED582A4BDEDD6E9EDC951E5B37AAA6D
    Sophia.Script.v5.10.2.zip 82567817994E4573FD39801B804C4B9AA8FD9E1DC14450041E6BEE14545549CF
    Sophia.Script.Wrapper.zip F9FE14E6388A1E96B8C067F184D93285C6C94E2474DAC4AC5E818F2C7C3732C1

A simple `(Get-FileHash -Path "path/to/archive.zip" -Algorithm SHA256).Hash > SHA256SUM` would include only hashes but not file name in `SHA256SUM` (and PowerShell returns a *path* instead of *filename* with `Get-FileHash`). So I don't know how to do that without external tools in PowerShell.

**Reply by farag2** [reply]
Ah. I'll check it. PS can do it too))

**Reply by farag2** [reply]
Uploaded SHA256SUM!

**Reply by farag2** [reply]
> So I don't know how to do that without external tools in PowerShell.

```powershell
<#
	.SYNOPSIS
	Calculate SHA256 hash and save it into a SHA256SUM file with UTF-8 encoding

	.PARAMETER Path
	Path to a foder with downloaded Sophia Script's zip archives

	.EXAMPLE
	SHA256SUM -Path D:\Sophia
#>
function SHA256SUM
{
	param
	(
		[string]
		$Path
	)

	Get-ChildItem -Path $Path -Filter Sophia.Script*.zip -Force | ForEach-Object -Process {
		"$($_.Name) $((Get-FileHash -Path $_.FullName -Algorithm SHA256).Hash)"
	} | Add-Content -Path "$Path\SHA256SUM" -Encoding Default -Force
}
```

**Reply by ghost** [reply]
Thanks for the feedback, the code snippet is quite useful!

**Comment by pifiuk** [comment]
For some reason in this new release, the Run Powershell button is greyed out. I tried reverting back to the last script but it just says that a new version is available and exits the script.

**Reply by farag2** [reply]
Can you provide a screenshot? It seems you're taking about the wrapper. Did you import the preset file first?

**Reply by pifiuk** [reply]
Yes sorry, generally I extract the wrapper files into the script folder. See attached. 

![sophia](https://user-images.githubusercontent.com/18093676/115887977-58d7b980-a44a-11eb-8498-c057356716af.jpg)

**Reply by farag2** [reply]
You an run it now wherever you want. Re-download the new version. I attached it.
